custom_components/moonraker/__init__.py

```diff
@@ -142,10 +142,23 @@ async def _async_fetch_data(self, query_path, query_object):
         except Exception as exception:
             raise UpdateFailed() from exception
 
+    async def _async_post(self, query_path) -> None:
+        if not self.moonraker.client.is_connected:
+            _LOGGER.warning("connection to moonraker down, restarting")
+            await self.moonraker.start()
+        try:
+            await self.moonraker.client.call_method(query_path)
+        except Exception as exception:
+            raise UpdateFailed() from exception
+
     async def async_get_cameras(self):
         """Return list of cameras"""
         return await self._async_fetch_data("server.webcams.list", None)
 
+    async def emergency_stop(self):
```

Comment:
This is too specific to be created here IMO, Coordinator should have least amount logic. 

If we need to do the post `_async_post` that is good, but the emergency stop is too specific imo

Reply:
done